### ISSUE TITLE
Automated testing for Snapshot integrations

### ISSUE THREAD

**Issue by jnaviask** [hicommonwealth/commonwealth#3557]
## Description
Related to #3347. We should come up with a plan to build unit/integration/e2e testing for Snapshot. We will likely either need some sort of testing server/apparatus, or else detailed mocks for snapshot cases.

## Acceptance Criteria
- Snapshot integrations have automated tests associated with them.

**Comment by jnaviask** [comment]
@alexyoung23j: feel free to consult with @mhagel, @NakulManchanda, and @ianrowan on this ticket, as they've recently implemented similar testing libraries for Cosmos / ETH. There will likely be differences in implementation, as Snapshot is not a blockchain, but they can help you understand the scope of work involved in setting up a testing fixture.

**Comment by NakulManchanda** [comment]
yes, do let me know about writing any kind of automated test or set up infrastructure on local or on heroku.
Thanks!

**Comment by NakulManchanda** [comment]
@alexyoung23j @ianrowan @mhagel 

**unit/integration/e2e testing**
On backend server side - we have place to write all three, we just have to capture existing behavior before we make a switch.
After @ianrowan we can run same tests again on backend

Current coverage on backend for `Snapshot tests`
`subscriptions.spec.ts`  have few test
`describe('SnapshotProposal /createSubscription test', () => {`
`describe('/viewNotification Snapshot Proposals', () => {`


On frontend, coverage anyways is little light:
![image](https://user-images.githubusercontent.com/4791635/234112026-9a1c489b-5501-44c5-afd9-65017bc4c42d.png)

We can add new `snapshots.spec.ts`
We don't have mocking api response support on frontend.
There are two ways i found for mocking responses for frontend:
- mocking xhr calls
- using msw - https://mswjs.io/ 

eg.
- xhr mcking - https://gist.github.com/NakulManchanda/0ed81ae37788f20f3a562fb787289984
-  MSW: https://gist.github.com/NakulManchanda/040a03c9da44c6b5206396d0a84a958d

We should start with mocking `/snapshots` or `/cosmos` endpoints only

**Comment by NakulManchanda** [comment]
FYI @kurtisassad - see comment above
https://github.com/hicommonwealth/commonwealth/issues/3557#issuecomment-1520805794

It may help closing blocked PR on testing frontend controller like:
https://github.com/hicommonwealth/commonwealth/pull/3391
